crates/edr_eth/src/remote/client.rs

```diff
@@ -969,23 +814,19 @@ impl RpcClient {
             .await
     }
 
-    /// Methods for retrieving multiple transaction receipts in one batch
+    /// Methods for retrieving multiple transaction receipts using concurrent
+    /// requests.
     #[cfg_attr(feature = "tracing", tracing::instrument(level = "trace", skip(self)))]
     pub async fn get_transaction_receipts(
         &self,
         hashes: impl IntoIterator<Item = &B256> + Debug,
     ) -> Result<Option<Vec<BlockReceipt>>, RpcClientError> {
-        let requests: Vec<RequestMethod> = hashes
+        let requests = hashes
             .into_iter()
-            .map(|transaction_hash| RequestMethod::GetTransactionReceipt(*transaction_hash))
-            .collect();
+            .map(|transaction_hash| self.get_transaction_receipt(transaction_hash));
```

Comment:
Doesn't clippy complain about this?

Can't it be `map(RpcClient::get_transaction_receipt)`?

Reply:
My understanding is: `get_transaction_receipt` is a method, so it can't be called as `RpcClient::get_transaction_receipt`. And [apparently](https://users.rust-lang.org/t/using-method-directly-instead-of-writing-lambda/85442/7) Rust doesn't support methods in those scenarios, so you can't do `.map(self.get_transaction_receipt)`.

Reply:
The syntax for doing `.map(self.get_transaction_receipt` is `RpcClient::get_transaction_receipt`, as the functions first argument is `self`

Reply:
`.map(RpcClient::get_transaction_receipt)` doesn't compile though

Reply:
Sorry, I updated my response. I overlooked the two arguments and noticed that when I tried it locally